scene/gui/text_edit.h

```diff
@@ -636,6 +638,15 @@ class TextEdit : public Control {
 	void set_right_click_moves_caret(bool p_enable);
 	bool is_right_click_moving_caret() const;
 
+	Selection::Mode get_selection_mode() { return selection.selecting_mode; };
```

Comment:
Need `const` on the getters. eg `Selection::Mode get_selection_mode() const;` Would also move them to the cpp file rather then keeping them here in the header.